src/emitters/constant.cpp

```diff
@@ -62,20 +65,29 @@ class ConstantBackgroundEmitter final : public Emitter<Float, Spectrum> {
                                           Mask active) const override {
         MTS_MASKED_FUNCTION(ProfilerPhase::EndpointSampleRay, active);
 
-        // 1. Sample spectrum
-        auto [wavelengths, weight] = m_radiance->sample_spectrum(
-            ek::zero<SurfaceInteraction3f>(),
-            math::sample_shifted<Wavelength>(wavelength_sample), active);
-
-        // 2. Sample spatial component
+        // 1. Sample spatial component
         Vector3f v0 = warp::square_to_uniform_sphere(sample2);
+        Point3f origin = m_bsphere.center + v0 * m_bsphere.radius;
 
-        // 3. Sample directional component
+        // 2. Sample directional component
         Vector3f v1 = warp::square_to_cosine_hemisphere(sample3);
-
-        return std::make_pair(Ray3f(m_bsphere.center + v0 * m_bsphere.radius,
-                                    Frame3f(-v0).to_world(v1), time, wavelengths),
-                              unpolarized<Spectrum>(weight) * (4.f * ek::sqr(ek::Pi<Float> * m_bsphere.radius)));
+        Vector3f direction = Frame3f(-v0).to_world(v1);
+
+        // 3. Sample spectrum
+        // TODO: how to best construct this `si`?
+        SurfaceInteraction3f si;
```

Comment:
You will probably have to zero-initialize this, or the GPU version will throw an exception in the virtual function call.
Actually, I am wondering if we can perhaps have `sample_wavelengths` *not* depend on the specific position?

Reply:
Since `sample_wavelengths` returns a sampling weight that includes the emitter's radiance, it needs a position for textured emitters I think?